src/main/java/ru/savin/minicrm/controller/RegistrationController.java

```
@@ -53,22 +52,26 @@ public String processRegistrationForm(
             return "/user/registration-form";
         }
 
-        User existingByUsername = userService.findByUserName(userName);
+        User existingByUsername = userService.findByUserName(userName); // todo change findByUserName to return Optional
         if (existingByUsername != null) {
             theModel.addAttribute("formUser", new FormUser());
             theModel.addAttribute("registrationError", "User name already exists.");
 
-            logger.warning("User with the name " + formUser.getUserName() + " already exists.");
+            logger.warning("User with the name " + formUser.getUserName() + " already exists."); // todo warning for why?
             return "/user/registration-form";
         }
 
-        User existingByEmail = userService.findByEmail(formUser.getEmail());
+        User existingByEmail = userService.findByEmail(formUser.getEmail()); // todo change findByEmail to return Optional
         if (existingByEmail != null) {
             theModel.addAttribute("registrationErrorEmail", "User with the email " + formUser.getEmail() + " already exists.");
 
-            logger.warning("User with the email " + formUser.getEmail() + " already exists.");
+            logger.warning("User with the email " + formUser.getEmail() + " already exists."); // todo warning for why?
             return "/user/registration-form";
         }
+        // todo все проверки лучше вынести в другой класс и каждую проверку в отдельный метод.
```

мне кажется я такой говнокод лютый написал для этого. Сделать-то все это просто, но я засомневался, передавать мне параметры в эти методы проверки или сделать в этом классе с проверками поля и передать их через конструктор. В первом случае мне можно сделать что-то типо утилитного класса со статическими методами, во втором - для проверки нужно создавать объект и в конструктор передавать параметры.
Короче, мне просто надо понять, как правильней поступить.

---

ну.. вполне такой валидатор класс можно сделать со статическими методами, думаю не обязательно иметь инстансы разные, ведь по сути никакого состояния он хранить не должен, просто валидирует параметры. Так что делай статик класс, добавляй статик методы, в каждый прокидывай нужные поля. Если понадобится, сделай общий метод коллектор, который принимает все, и выдает общий вердикт. Вообще тут все зависит от стратегий валидации, я рекомендую тебе сделать по стратегии `fail fast` - это упасть на валидации при первом же некорректном параметре. поэтому можно сделать без объединяющего метода коллектора